### PR TITLE
migrated MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO

### PR DESCRIPTION
**Pull request by Kunj125**
## Related Issues/PRs

 #9228

## What changes are proposed in this pull request?

Migrate `MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO`

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/environment_variables.py [hunk 1]
```diff
@@ -346,6 +346,10 @@ def get(self):
     "MLFLOW_ENV_ROOT", str, str(Path.home().joinpath(".mlflow", "envs"))
 )
 
+#: Specifies whether or not to use DBFS FUSE mount to store artifacts on Databricks 
+#: (default: ``False``)
+MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO = _EnvironmentVariable("MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO", str, False)
```
Comment (harupy) [inline mlflow/environment_variables.py:351]:
```suggestion
MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO = _EnvironmentVariable("MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO", str,"")
```

Comment (harupy) [reply mlflow/environment_variables.py:351]: This env var looks like a flag. Let's use `_BooleanEnvironmentVariable`.

Comment (harupy) [reply mlflow/environment_variables.py:351]:
```suggestion
MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO = _BooleanEnvironmentVariable("MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO", False)
```

File: mlflow/store/artifact/dbfs_artifact_repo.py [hunk 2]
```diff
@@ -213,7 +212,7 @@ def dbfs_artifact_repo_factory(artifact_uri):
         return DatabricksArtifactRepository(cleaned_artifact_uri)
     elif (
         mlflow.utils.databricks_utils.is_dbfs_fuse_available()
-        and os.environ.get(USE_FUSE_ENV_VAR, "").lower() != "false"
+        and MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO.get().lower() != "false"
```
Comment (harupy) [inline mlflow/store/artifact/dbfs_artifact_repo.py:215]: With https://github.com/mlflow/mlflow/pull/9250/files#r1286953735, we can fix this line like this:

```suggestion
        and MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO.get()
```

Comment (Kunj125) [reply mlflow/store/artifact/dbfs_artifact_repo.py:215]: Yeah, this is better! Thanks.

**Comment by harupy** [comment]
@Kunj125 You can use the following command to squash commits:

```
git reset $(git merge-base master $(git branch --show-current))
git commit -s -m "<message>"
```

File: mlflow/environment_variables.py [hunk 3]
```diff
@@ -100,55 +103,65 @@ def get(self):
...
-MLFLOW_GCS_DOWNLOAD_CHUNK_SIZE = _EnvironmentVariable("MLFLOW_GCS_DOWNLOAD_CHUNK_SIZE", int, None)
+MLFLOW_GCS_DOWNLOAD_CHUNK_SIZE = _EnvironmentVariable(
+    "MLFLOW_GCS_DOWNLOAD_CHUNK_SIZE", int, None)
 
 #: Specifies the chunk size to use when uploading a file to GCS.
 #: (default: ``None``). If None, the chunk size is automatically determined by the
 #: ``google-cloud-storage`` package.
-MLFLOW_GCS_UPLOAD_CHUNK_SIZE = _EnvironmentVariable("MLFLOW_GCS_UPLOAD_CHUNK_SIZE", int, None)
+MLFLOW_GCS_UPLOAD_CHUNK_SIZE = _EnvironmentVariable(
+    "MLFLOW_GCS_UPLOAD_CHUNK_SIZE", int, None)
```
Comment (harupy) [inline mlflow/environment_variables.py:125]: Can you avoid unrelated changes?

Comment (Kunj125) [reply mlflow/environment_variables.py:125]: I am trying to pass the DCO check. I am stuck in rebasing.

**Comment by harupy** [comment]
I think you can rebase now:

```
git rebase master
git push -f origin HEAD
```

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by Kunj125** [comment]
> LGTM!

Thanks @harupy for your help!

**Comment by harupy** [comment]
@mlflow-automation autoformat

File: mlflow/environment_variables.py [hunk 4]
```diff
@@ -346,6 +346,12 @@ def get(self):
     "MLFLOW_ENV_ROOT", str, str(Path.home().joinpath(".mlflow", "envs"))
 )
 
+#: Specifies whether or not to use DBFS FUSE mount to store artifacts on Databricks
+#: (default: ``False``)
+MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO = _BooleanEnvironmentVariable(
+    "MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO", False
```
Comment (harupy) [inline mlflow/environment_variables.py:352]:
```suggestion
    "MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO", True
```
